source/RemoteWiring/RemoteDevice.cpp

```diff
@@ -159,6 +180,7 @@ RemoteDevice::pinMode(
 	uint8_t port_mask;
 	getPinMap( pin_, &port, &port_mask );
 
+    _firmata->lock();
```

Comment:
You may be able to use a lock guard instead of this. Something like
```std::lock_guard<Firmata::UwpFirmata> lock(*_firmata);```

http://en.cppreference.com/w/cpp/thread/lock_guard